src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedBackingFieldSymbol.cs

```diff
@@ -72,15 +77,165 @@ internal override bool HasPointerType
             }
         }
 
+        public override ImmutableArray<CSharpAttributeData> GetAttributes()
+            => GetAttributesBag().Attributes;
+
+        IAttributeTargetSymbol IAttributeTargetSymbol.AttributesOwner
+            => this;
+
+        AttributeLocation IAttributeTargetSymbol.AllowedAttributeLocations
+            => AttributeLocation.Field;
+
+        AttributeLocation IAttributeTargetSymbol.DefaultAttributeLocation
+            => AttributeLocation.Field;
+
+        internal override Location ErrorLocation
+            => _property.Location;
+
+        internal sealed override int? TypeLayoutOffset
+            => GetDecodedWellKnownAttributeData()?.Offset;
+
+        internal sealed override MarshalPseudoCustomAttributeData MarshallingInformation
+            => GetDecodedWellKnownAttributeData()?.MarshallingInformation;
+
         internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, ref ArrayBuilder<SynthesizedAttributeData> attributes)
         {
             base.AddSynthesizedAttributes(moduleBuilder, ref attributes);
 
             var compilation = this.DeclaringCompilation;
 
-            // Dev11 doesn't synthesize this attribute, the debugger has a knowledge 
+            // Dev11 doesn't synthesize this attribute, the debugger has a knowledge
             // of special name C# compiler uses for backing fields, which is not desirable.
             AddSynthesizedAttribute(ref attributes, compilation.SynthesizeDebuggerBrowsableNeverAttribute());
         }
+
+        internal sealed override bool IsNotSerialized
```

Comment:
>IsNotSerialized [](start = 38, length = 15)

It feels like a lot of code duplicated here, I think we should find a way to share it instead. Perhaps we should stop inheriting from SynthesizedFieldSymbolBase and introduce a new base class, from which SourceFieldSymbol and SynthesizedBackingFieldSymbol can inherit.

Reply:
That worked out better indeed. Thanks Aleksey